Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -617,22 +650,38 @@ public void refreshControls()
             getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT ?
                     "-Portrait" : "";
 
-    // Add all the enabled overlay items back to the HashSet.
-    if (EmulationActivity.isGameCubeGame() || mPreferences.getInt("wiiController", 3) == 0)
+    if (mPreferences.getBoolean("showInputOverlay", true))
     {
-      addGameCubeOverlayControls(orientation);
-    }
-    else if (mPreferences.getInt("wiiController", 3) == 4)
-    {
-      addClassicOverlayControls(orientation);
-    }
-    else
-    {
-      addWiimoteOverlayControls(orientation);
-      if (mPreferences.getInt("wiiController", 3) == 3)
+      // Add all the enabled overlay items back to the HashSet.
+      if (EmulationActivity.isGameCubeGame() || mPreferences.getInt("wiiController", 3) == 0)
+      {
+        addGameCubeOverlayControls(orientation);
+      }
+      else if (mPreferences.getInt("wiiController", 3) == 4)
       {
-        addNunchukOverlayControls(orientation);
+        addClassicOverlayControls(orientation);
       }
+      else
+      {
+        addWiimoteOverlayControls(orientation);
+        if (mPreferences.getInt("wiiController", 3) == 3)
+        {
+          addNunchukOverlayControls(orientation);
+        }
+      }
+    }
+
+    if (!EmulationActivity.isGameCubeGame())
+    {
+      int doubleTapButton = mPreferences.getInt("doubleTapButton",
+              InputOverlayPointer.DOUBLE_TAP_OPTIONS.get(InputOverlayPointer.DOUBLE_TAP_A));
+
+      if (mPreferences.getInt("wiiController", OVERLAY_WIIMOTE_NUNCHUCK) !=
+              InputOverlay.OVERLAY_WIIMOTE_CLASSIC &&
+              doubleTapButton == InputOverlayPointer.DOUBLE_TAP_CLASSIC_A)
+        doubleTapButton = InputOverlayPointer.DOUBLE_TAP_A;
```

Comment:
With the condition spanning multiple lines, this should probably have some curlies around.